### ISSUE TITLE
the `deep` option does not work

### ISSUE THREAD

**Issue by zoubin** [mrmlnc/fast-glob#129]
### Environment

  * OS Version: macOS 10.13.6
  * Node.js Version: 10.11.0

### Actual behavior
When `options.deep` is set to 1, items with depth 2 are also returned

### Expected behavior
Only items with depth less than 2 should be returned

### Steps to reproduce

```bash
npm i fast-glob
mkdir -p test/fixture/foo
touch test/fixture/foo/index.js
node repro.js
```

### Code sample
repro.js

```js
const fg = require('fast-glob')
fg('test/**/*.js', { deep: 1 }).then(console.log)

```

### More
I believe is due to some lines of code in https://github.com/JS-DevTools/readdir-enhanced/blob/master/lib/directory-reader.js#L187

```js
        if (this.shouldRecurse(stats, posixPath, maxDepthReached)) {
          // Add this subdirectory to the queue
          this.queue.push({
            path: fullPath,
            basePath: itemPath + options.sep,
            posixBasePath: posixPath + '/',
            depth: dir.depth + 1,
          });
        }

        // Determine whether this item matches the filter criteria
        if (this.filter(stats, posixPath)) {
          this.pushOrBuffer({
            data: options.stats ? stats : itemPath,
            file: stats.isFile(),
            directory: stats.isDirectory(),
            symlink: stats.isSymbolicLink(),
          });
        }
```
Though `this.shouldRecurse` returns `false`, `this.filter` does return `true`.

**Comment by mrmlnc** [comment]
Should work correctly after #132.

**Comment by mrmlnc** [comment]
Will be shipped at #136.
